### PR TITLE
Switch to dart-sass

### LINKED ISSUE #3900: Drop support for node-sass
This involves:
- make building with dart sass default
- find a way to pass version to SASS from dart sass (or ignore this feature) - it is related to functions.js
- update the documentation (remove node sass from the docs)
- update the code to fix warnings (division in dart sass)

### PR DESCRIPTION
**Pull request by sowasred2012**
Co-authored-by: Steve Rydz <steve.rydz@canonical.com>

## Done

- Switches out `node-sass` for `sass` package, updates SCSS build scripts accordingly.
- Removes `node-sass` build information from the docs
- Removed the version number function script
  - `sass` doesn't support passing a JS file with custom functions, open to suggestions on ways around this. For now, the version number is hardcoded.
- Fixes warnings relating to slash divider / math.div
- Updated github actions accordingly

### Drive bys
This being the first PR targeting the `vanilla-3.0` branch, we initially updated the Vanilla version to 3.0 everywhere as well as switching to `dart-sass`. We fixed some things that were broken by the exclusion of deprecated styles before realising we were biting off more than we could chew in the scope of this PR (and that these changes should be addressed in https://github.com/canonical-web-and-design/vanilla-framework/issues/3744), so reverted the version number to 2.0, and kept the changes we'd made:
- Replaced the use of `heading-max-width--short` et al mixins with the suggested `map-get($max-widths, default)`
- Fixed some deprecated mark up in the vf.io site navigation

Fixes #3900 & #3756

## QA

- Check the PR builds, and the demo runs

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [component status page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/component-status.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: scss/_base_forms-range.scss [hunk 1]
```diff
@@ -37,7 +37,7 @@
       border-radius: $thumb-radius;
       box-shadow: $thumb-shadow;
       height: $thumb-size;
-      margin-top: ((-$thumb-size + $track-height) / 2) - $track-border-size;
+      margin-top: math.div((-$thumb-size + $track-height), 2) - $track-border-size;
```
Comment (bartaz) [inline scss/_base_forms-range.scss:40]: In cases like these when the math is simple I'd change it to `* 0.5` to avoid function calls.

```suggestion
      margin-top: ((-$thumb-size + $track-height) * 0.5) - $track-border-size;
```

File: scss/_base_forms-range.scss [hunk 2]
```diff
@@ -80,10 +80,10 @@
     &::-ms-track {
       background: transparent;
       border-color: transparent;
-      border-width: $thumb-size / 2;
+      border-width: math.div($thumb-size, 2);
```
Comment (bartaz) [inline scss/_base_forms-range.scss:83]:
```suggestion
      border-width: $thumb-size * 0.5;
```

File: scss/_base_forms-range.scss [hunk 3]
```diff
@@ -80,10 +80,10 @@
...
       color: transparent;
       height: $track-height + 1;
-      width: calc(100% - (#{$thumb-size} / 2));
+      width: calc(100% - (math.div(#{$thumb-size}, 2)));
```
Comment (bartaz) [inline scss/_base_forms-range.scss:86]:
```suggestion
      width: calc(100% - (#{$thumb-size} * 0.5));
```

File: scss/_patterns_notifications.scss [hunk 4]
```diff
@@ -100,7 +100,7 @@ $notification-text-margin-bottom: $spv-outer--medium - $spv-nudge;
       background-size: unset;
       border: 0;
       position: absolute;
-      right: $sph-inner / 2;
+      right: math.div($sph-inner, 2);
```
Comment (bartaz) [inline scss/_patterns_notifications.scss:103]:
```suggestion
      right: $sph-inner * 0.5;
```

**Review by bartaz — COMMENTED** [review]
Some suggestions on making some math simpler.

**Comment by bartaz** [comment]
@sowasred2012 The `watch:scss:skip-standalone` is to allow quicker builds by skiping standalone stylesheets (during development). It seems that with dart sass being slower it may become even more useful.

File: scss/_settings_system.scss [hunk 5]
```diff
@@ -1,7 +1 @@
-// Global system settings
-@if function-exists(vanilla-version) {
-  $app-version: vanilla-version();
-} @else {
-  // update the fallback with major releases
-  $app-version: '2.0.0' !default;
-}
+$app-version: '2.0.0' !default;
```
Comment (bartaz) [inline scss/_settings_system.scss:1]: It's a 3.0 branch, so why reverting to 2.0?

File: templates/_layouts/_header.html [hunk 6]
```diff
@@ -15,9 +15,9 @@
       <ul class="p-navigation__items">
         <li class="p-navigation__item {% if path.startswith('/docs') and not path.startswith('/docs/examples') %}is-selected{% endif %}"><a class="p-navigation__link" href="/docs">Docs</a></li>
         <li class="p-navigation__item {% if '/docs/examples' in path %}is-selected{% endif %}"><a class="p-navigation__link" href="/docs/examples">Examples</a></li>
-        <li class="p-navigation__link{% if path == '/accessibility' %} is-selected{% endif %}"><a href="/accessibility">Accessibility</a></li>
-        <li class="p-navigation__link{% if path == '/browser-support' %} is-selected{% endif %}"><a href="/browser-support">Browser support</a></li>
-        <li class="p-navigation__link{% if path == '/contribute' %} is-selected{% endif %}"><a href="/contribute">Contribute</a></li>
+        <li class="p-navigation__item{% if path == '/accessibility' %} is-selected{% endif %}"><a class="p-navigation__link" href="/accessibility">Accessibility</a></li>
```
Comment (bartaz) [inline templates/_layouts/_header.html:18]: Also I'm not sure why this is in this PR. It's fine if drive-by, but lets list it in the description (and link to fixed issues if we have them).

Comment (sowasred2012) [reply templates/_layouts/_header.html:18]: There wasn't an issue for this, it was spotted when we initially switched the version to 3.0 and some deprecated styles were excluded. In any case, it's old markup that should have been updated, so I'll list it as an item in the description

File: scss/_base_typography-definitions.scss [hunk 7]
```diff
@@ -3,14 +3,13 @@
 @mixin vf-b-typography-definitions {
   //@section Heading styling in placeholders
   %vf-heading-1 {
-    @include heading-max-width--short;
```
[the remark this answers is not in the document]
Comment (sowasred2012) [reply scss/_base_typography-definitions.scss:6]: Just a drive by - I'll make a note

**Review by bartaz — APPROVED** [review]
LGTM thanks

@sowasred2012 Knowing that removing deprecated stuff should we maybe revisit the estimate on the "remove deprecated" task?
